lightning/src/routing/router.rs

```diff
@@ -1785,6 +1785,49 @@ fn add_random_cltv_offset(route: &mut Route, payment_params: &PaymentParameters,
 	}
 }
 
+/// Build a route from us (payer) with the given hops ending at the target node (payee).
+///
+/// Re-uses logic from `find_route`, so the restrictions described there also apply here.
+pub fn build_route_from_hops<L: Deref>(
+	our_node_pubkey: &PublicKey, hops: &[PublicKey], route_params: &RouteParameters, network: &NetworkGraph,
+	logger: L, random_seed_bytes: &[u8; 32]) -> Result<Route, LightningError>
+where L::Target: Logger {
+	let network_graph = network.read_only();
+	match build_route_from_hops_internal(
+		our_node_pubkey, hops, &route_params.payment_params, &network_graph,
+		route_params.final_value_msat, route_params.final_cltv_expiry_delta, logger, random_seed_bytes
+		) {
+		Ok(mut route) => {
+			add_random_cltv_offset(&mut route, &route_params.payment_params, &network_graph, random_seed_bytes);
+			Ok(route)
+		},
+		Err(err) => Err(err),
```

Comment:
nit: little more idiomatic as

```rust
let mut route = build_route_from_hops_internal(...)?;
```

Reply:
Adressed this in https://github.com/lightningdevkit/rust-lightning/pull/1491/commits/19ecb4c02798caf3d89fd42e9497c78565efe756, also changed `find_route` correspondingly.